pkg/provider/azure_controller_common.go

```diff
@@ -677,3 +724,26 @@ func isInstanceNotFoundError(err error) bool {
 	}
 	return strings.Contains(errMsg, errStatusCode400) && strings.Contains(errMsg, errInvalidParameter) && strings.Contains(errMsg, errTargetInstanceIds)
 }
+
+func getErrorCode(err error) *string {
+	if err == nil {
+		return nil
+	}
+	r := regexp.MustCompile(`Code="(.*?)".*`)
```

Comment:
This should be a global variable rather than compiling the regex each time an error code needs to be parsed.

Reply:
Ah you are right. Good catch 👍